wallet/wallet-rpc-lib/src/rpc/types.rs

```diff
@@ -359,26 +360,19 @@ impl NftMetadata {
     }
 }
 
-#[derive(Debug, Copy, Clone, serde::Serialize, serde::Deserialize, HasValueHint)]
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, HasValueHint)]
 pub enum TokenTotalSupply {
-    Fixed(DecimalAmount),
+    Fixed(RpcAmountIn),
     Lockable,
     Unlimited,
 }
 
 impl TokenTotalSupply {
-    fn into_token_supply<N: NodeInterface>(
-        self,
-        chain_config: &ChainConfig,
-    ) -> Result<tokens::TokenTotalSupply, RpcError<N>> {
+    fn to_token_supply(&self, decimals: u8) -> Option<tokens::TokenTotalSupply> {
         match self {
-            Self::Lockable => Ok(tokens::TokenTotalSupply::Lockable),
-            Self::Unlimited => Ok(tokens::TokenTotalSupply::Unlimited),
-            Self::Fixed(amount) => {
-                let decimals = chain_config.coin_decimals();
-                let amount = amount.to_amount(decimals).ok_or(RpcError::InvalidCoinAmount)?;
-                Ok(tokens::TokenTotalSupply::Fixed(amount))
-            }
+            Self::Lockable => Some(tokens::TokenTotalSupply::Lockable),
+            Self::Unlimited => Some(tokens::TokenTotalSupply::Unlimited),
+            Self::Fixed(amount) => amount.to_amount(decimals).map(tokens::TokenTotalSupply::Fixed),
```

Comment:
This had `chain_config` passed to it and used coin decimals t parse the string. To me it looks like a bug: the number of decimal places specified by the token should be used here, so I changed it. Seems no tests picked up on it.

@OBorce could you please check this is correct?

Reply:
yes this is correct, the token functional tests are not yet running on the rpc wallet I will try to migrate them